swiftide/src/integrations/qdrant/indexing_node.rs

```diff
@@ -49,11 +49,117 @@ impl TryInto<qdrant::PointStruct> for Node {
             .collect::<HashMap<&str, Value>>()
             .into();
 
+        let Some(vectors) = self.vectors else {
+            bail!("Node without vectors")
+        };
+        let vectors = try_create_vectors(vectors)?;
+
         // Construct the `qdrant::PointStruct` and return it.
-        Ok(qdrant::PointStruct::new(
-            id,
-            self.vector.context("Vector is not set")?,
-            payload,
-        ))
+        Ok(qdrant::PointStruct::new(id, vectors, payload))
+    }
+}
+
+fn try_create_vectors(vectors: HashMap<EmbeddableType, Vec<f32>>) -> Result<qdrant::Vectors> {
+    if vectors.is_empty() {
+        bail!("Node with empty vectors")
+    } else if vectors.len() == 1 {
+        let vector = vectors.into_values().next().expect("Node has vector entry");
```

Comment:
I did just panic on `next()` because it should never happen thanks to `vectors.len() == 1` condition. I will change it to `bail!` because it looks wrong.